cmd/minishift/cmd/root.go

```diff
@@ -117,14 +128,49 @@ var RootCmd = &cobra.Command{
 			}
 		}
 
-		ensureConfigFileExists(constants.ConfigFile)
+		// Check if old instance state config file exists and move it to new one.
+		// Create MACHINE_NAME-state.json (machine state config file)
+		_, err = os.Stat(minishiftConstants.GetInstanceStateConfigOldPath())
+		if err == nil {
+			if err := os.Rename(minishiftConstants.GetInstanceStateConfigOldPath(), minishiftConstants.GetInstanceStateConfigPath()); err != nil {
+				atexit.ExitWithMessage(1, fmt.Sprintf("Error moving old state config to new one: %s", err.Error()))
+			}
+		}
 
-		// Create MACHINE_NAME.json
+		minishiftConfig.InstanceStateConfig, err = minishiftConfig.NewInstanceStateConfig(minishiftConstants.GetInstanceStateConfigPath())
+		if err != nil {
+			atexit.ExitWithMessage(1, fmt.Sprintf("Error creating config for VM: %s", err.Error()))
+		}
+
+		// Create MACHINE_NAME.json (machine config file)
 		minishiftConfig.InstanceConfig, err = minishiftConfig.NewInstanceConfig(minishiftConstants.GetInstanceConfigPath())
 		if err != nil {
 			atexit.ExitWithMessage(1, fmt.Sprintf("Error creating config for VM: %s", err.Error()))
 		}
 
+		// Check if there is already something in hostfolder from old config file and write it to new one.
+		// This should be removed after 2-3 release of minishift.
+		hostfolder := minishiftConfig.InstanceStateConfig.HostFolders
+		if len(hostfolder) != 0 {
+			minishiftConfig.InstanceConfig.HostFolders = hostfolder
+			if err != minishiftConfig.InstanceConfig.Write() {
+				atexit.ExitWithMessage(1, fmt.Sprintf("Error coping existing hostfolder to new instance config: %s", err.Error()))
+			}
+		}
+
+		// Check if there is already something in cacheImages from old config file and write it to new one.
+		// This should be removed after 2-3 release of minishift.
```

Comment:
`If cache-config exists then copy it to the new instance config. This change should be removed after few releases as we expect our users will not need this.`